packages/lesswrong/server/notificationCallbacks.tsx

```diff
@@ -344,6 +344,20 @@ function PostsCurateNotification (post, oldPost) {
 }
 addCallback("posts.edit.async", PostsCurateNotification);
 
+async function TaggedPostNewNotifications(tagRel) {
```

Comment:
Somewhere in this function you need to check that the notified user has access to the post, since drafts can be tagged. You may also need an additional callback for when a tagged post is undrafted, for that reason.

Reply:
Hmm. I notice also that currently (afaict?) if I receive a notification for a new post (from a regular "subscribe to user" subscription), and then later that post is moved to draft, I still have access to it in my notifications sidebar.

I think this means notifications are generally not very secure right now.

Reply:
Update: I'd forgotten that the user I was testing with was in sunshineRegiment. When I remove that, I can still see the notification, but it doesn't have a hover-preview. Less bad but still not great.